### PR TITLE
fix: [PIE-1338]: Adding support for new variable creator in http step

### PR DESCRIPTION
**Pull request by archit-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32104)

### CODE REVIEW

File: 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java [hunk 1]
```diff
@@ -35,9 +37,10 @@
 @OwnedBy(PIPELINE)
 @RecasterAlias("io.harness.plancreator.steps.http.HttpStepNode")
 public class HttpStepNode extends PmsAbstractStepNode {
-  @JsonProperty("type") @NotNull StepType type = StepType.Http;
+  @JsonProperty("type") @NotNull @VariableExpression(replaceWithUUid = false) StepType type = StepType.Http;
```
Comment (hingerabhinav) [inline 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java:40]: @archit-harness why is type a VariableExpression?

Comment (archit-harness) [reply 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java:40]: stage.steps.step1.type is also an expression which can be used

**Review by vaibhav-si — CHANGES_REQUESTED** [review]
> Reviewed 13 of 19 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32104)* status: 19 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java, line 40 at r1](https://reviewable.io/reviews/harness/harness-core/32104#-N-1jX_Q5FZUbBGkTI5b:-N-1jX_Q5FZUbBGkTI5c:b-v6xtdd) ([raw file](https://github.com/harness/harness-core/blob/039c06409eccc11f6889d30d54893ed7ef58ae1d/860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java#L40)):*
> ```Java
> @RecasterAlias("io.harness.plancreator.steps.http.HttpStepNode")
> public class HttpStepNode extends PmsAbstractStepNode {
>   @JsonProperty("type") @NotNull @VariableExpression(replaceWithUUid = false) StepType type = StepType.Http;
> ```

replacewithUUid true for type field can be build in framework
___
*[860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java, line 43 at r1](https://reviewable.io/reviews/harness/harness-core/32104#-N-1jbJH9qU7GAKertVU:-N-1jbJH9qU7GAKertVV:bko9dr4) ([raw file](https://github.com/harness/harness-core/blob/039c06409eccc11f6889d30d54893ed7ef58ae1d/860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java#L43)):*
> ```Java
>   @JsonProperty("spec")
>   @JsonTypeInfo(use = NAME, property = "type", include = EXTERNAL_PROPERTY, visible = true)
>   @VariableExpression(policy = REGULAR_WITH_CUSTOM_FIELD, customFieldName = "spec")
> ```

we can use jsonproperty name, order will be 

field name - json property name - custom field name

File: 970-ng-commons/src/main/java/io/harness/http/HttpHeaderConfig.java [hunk 2]
```diff
@@ -19,5 +21,5 @@
 @Builder
 public class HttpHeaderConfig implements NestedAnnotationResolver {
   @Expression(ALLOW_SECRETS) String key;
-  @Expression(ALLOW_SECRETS) String value;
+  @Expression(ALLOW_SECRETS) @VariableExpression(policy = REGULAR_WITH_CUSTOM_FIELD) String value;
```
Comment (hingerabhinav) [inline 970-ng-commons/src/main/java/io/harness/http/HttpHeaderConfig.java:24]: So key does not have an fqn? And we can access header value through pipeline.stages.approval.spec.execution.steps.http.spec.headers.HEADER_KEY? Is that correct?

**Review by vaibhav-si — COMMENTED** [review]
we can use hidden true or not to know whether variable needs to be exposed or not. If not hidden, we can expose the variable by default. 

This would reduce the overhead for users.

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32104#-:-N-1kUQk4jgNIDXoGzmY:b-ia9zfi)* status: 19 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained.

File: 970-ng-commons/src/main/java/io/harness/http/HttpHeaderConfig.java (hunk 2 above)
Comment (archit-harness) [reply 970-ng-commons/src/main/java/io/harness/http/HttpHeaderConfig.java:24]: as its a list object, the framework handles includes key value in path, check HTTP step test

**Review by archit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32104)* status: 35 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java, line 40 at r1](https://reviewable.io/reviews/harness/harness-core/32104#-N-1jX_Q5FZUbBGkTI5b:-N-2qcAh9NhsnlPLErkF:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/039c06409eccc11f6889d30d54893ed7ef58ae1d/860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java#L40)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

replacewithUUid true for type field can be build in framework
</blockquote></details>

Done.
___
*[860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java, line 43 at r1](https://reviewable.io/reviews/harness/harness-core/32104#-N-1jbJH9qU7GAKertVU:-N-2qd1H99aOiauO3h-6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/039c06409eccc11f6889d30d54893ed7ef58ae1d/860-orchestration-steps/src/main/java/io/harness/plancreator/steps/http/HttpStepNode.java#L43)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

we can use jsonproperty name, order will be 

field name - json property name - custom field name
</blockquote></details>

Done.

**Review by vaibhav-si — COMMENTED** [review]
> Reviewed 3 of 19 files at r1, 23 of 24 files at r2, 1 of 1 files at r3, 5 of 6 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32104)* status: 40 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32104#-:-N-32nWwDql79TqEwYUP:bnfp4nl)* status: 40 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by prashantsharma04 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32104#-:-N-3QmFYD7KKYJsJlDQZ:bnfp4nl)* status: 40 file(s) to review. 2. unresolved discussions.

**Pull request merged**